ext/semian/semian.c

```diff
@@ -35,11 +35,13 @@ static ID id_timeout;
 static VALUE eSyscall, eTimeout, eInternal;
 static int system_max_semaphore_count;
 
-static const int kIndexTickets = 0;
-static const int kIndexTicketMax = 1;
-static const int kIndexLock = 2;
-
-static const int kNumSemaphores = 3;
+typedef enum
+{
+  sem_tickets,     // semaphore for the tickets currently issued
+  sem_max_tickets, // semaphore to track the maximum number of tickets available for issue
```

Comment:
I would almost consider renaming this one to `sem_configured_tickets`. I see it more as the value to indicate what the amount of currently configured tickets are, so we can accurately adjust.